packages/app-runtime/test/lib/MockUIBridge.matchers.ts

```diff
@@ -0,0 +1,122 @@
+import { MockUIBridge } from "./MockUIBridge";
+
+expect.extend({
+    showDeviceOnboardingCalled(mockUIBridge: unknown, deviceId: string) {
+        if (!(mockUIBridge instanceof MockUIBridge)) {
+            throw new Error("This method can only be used with expect(MockUIBridge).");
+        }
+
+        const calls = mockUIBridge.calls.filter((x) => x.method === "showDeviceOnboarding");
+        if (calls.length === 0) {
+            return { pass: false, message: () => "The method showDeviceOnboarding was not called." };
+        }
+
+        const matchingCalls = calls.filter((x) => x.deviceOnboardingInfo.id === deviceId);
+        if (matchingCalls.length === 0) {
+            return {
+                pass: false,
+                message: () =>
+                    `The method showDeviceOnboarding was called, but not with the specified device id '${deviceId}', instead with ids '${calls.map((e) => e.deviceOnboardingInfo.id).join(", ")}}'.`
```

Comment:
As far as I see it, there is one closing curly bracket too many at the end of the message.

Also, `${calls.map((e) => 'e.deviceOnboardingInfo.id').join(", ")}` instead of `'${calls.map((e) => e.deviceOnboardingInfo.id).join(", ")}'` should be more precise.

Reply:
Perhaps `'${e.deviceOnboardingInfo.id}'` is needed inside the mapping 🕵️‍♀️